packages/react-app/components/cards/MediaCard.jsx

```diff
@@ -69,140 +83,199 @@ function MediaCard({
 
   const toast = useToast();
 
+  const { isOpen, onOpen, onClose } = useDisclosure();
+
   const handleRequestPrivateProfileUnlock = async () => {
+    if (!stakeAmount || +stakeAmount < +"0.1") {
+      return toast({ title: "Please enter a valid stake amount. Minimum 0.1 MATIC.", status: "error" });
+    }
+    const weiStakeAmount = ethers.utils.parseEther(stakeAmount);
     try {
-      const tx = await dRecruitContract.request(privateProfile.tokenId, {
-        value: ethers.utils.parseEther("0.01"),
+      // Request allowance
+      const allowance = await tokenContract.allowance(context.address, dRecruitContract.address);
+      // Only ask for allowance if it is not enough
+      if (allowance.lt(weiStakeAmount) || allowance.lt(ethers.utils.parseEther("0.1"))) {
+        await tokenContract.approve(dRecruitContract.address, weiStakeAmount);
```

Comment:
Store this as a tx object and await it with `.wait()` We need to wait until the tx gets mined, not until it's sent.

Reply:
Done